style.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Vencord, a Discord client mod
+ * Copyright (c) 2024 Vendicated and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { findByProps } from "@webpack";
+
+const CssFormatCode: string = `body:has(
+    div.{sidebar}
+        > section
+        div.{wrapper}
+        div.{actionButtons}
+        > button:nth-child(2).{buttonActive}
+)
+.{messageContent} {
+filter: blur(12px);
+}
+
+body:has(
+    div.{sidebar}
+        > section
+        div.{wrapper}
+        div.{actionButtons}
+        > button:nth-child(2).{buttonActive}
+)
+.{visualMediaItemContainer} {
+filter: blur(50px) brightness(0.1);
+}
+
+body:has(
+    div.{sidebar}
+        > section
+        div.{wrapper}
+        div.{actionButtons}
+        > button:nth-child(2).{buttonActive}
+)
+.{embedWrapper} {
+filter: blur(50px);
+}
+
+body.vc-dnl-hide-in-streamer-mode:has(.{notice}.{colorStreamerMode})
+.{visualMediaItemContainer} {
+filter: blur(50px) brightness(0.1);
+}
+
+body.vc-dnl-hide-in-streamer-mode:has(.{notice}.{colorStreamerMode})
+.{messageContent} {
+filter: blur(12px);
+}
+
+body.vc-dnl-hide-in-streamer-mode:has(.{notice}.{colorStreamerMode})
+.{embedWrapper} {
+filter: blur(50px);
+}
+
+body.vc-dnl-show-messages .{visualMediaItemContainer} {
+filter: blur(0px) brightness(1) !important;
+}
+
+body.vc-dnl-show-messages .{messageContent} {
+filter: blur(0px) !important;
+}
+
+body.vc-dnl-show-messages .{embedWrapper} {
+filter: blur(0px) !important;
+}
+
+body.vc-dnl-hover-to-view .{messageContent}:hover {
+filter: blur(0px) brightness(1) !important;
+}
+
+body.vc-dnl-hover-to-view .{embedWrapper}:hover {
+filter: blur(0px) brightness(1) !important;
+}
+
+body.vc-dnl-hover-to-view .{visualMediaItemContainer}:hover {
+filter: blur(0px) brightness(1) !important;
+}`;
+
+/*
+[
+    "sidebar",
+    "wrapper",
+    "actionButtons",
+    "buttonActive",
+    "messageContent",
+    "visualMediaItemContainer",
+    "embedWrapper",
+    "notice",
+    "colorStreamerMode",
+]
+*/
+
+export function getStyle(): [string, object] {
+    const t = findByProps("messageEditorCompact"); // ["messageContent","wrapper"]
+    const v = findByProps("embedWrapper");
+    const u = findByProps("visualMediaItemContainer");
+    const l = findByProps("colorStreamerMode", "notice");
+    const o = findByProps("actionButtons", "buttonActive", "wrapper");
+    const f = findByProps("sidebar", "panels");
+    const Classes = Object.assign({}, o, l, u, v, t, f);
+    let CssCode = CssFormatCode;
+    for (const className in Classes) {
+        CssCode = CssCode.replaceAll(`{${className}}`, Classes[className]);
```

Comment:
not sure how I am able to use it if its giving an error in vscode `Property 'replaceAll' does not exist on type 'string'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2021' or later.ts(2550)` could be of how vencord is compiled

Reply:
That is a TS (guessing LSP) error I've had recently on other projects. The
code should be fine.